include/lbann/models/model.hpp

```diff
@@ -226,7 +232,8 @@ class model {
 
   /** The objective function used to train the model. */
   objective_function *m_objective_function;
-
+  /** Give model a name. */
+  std::string m_name;
```

Comment:
I kind of prefer default name to be empty string. Our print/timer callback is already too verbose for multiple instances (multi) model case (e.g., multi-model data parallel and LTFB) and in that case we don't need to name the model since it is already identified by its model rank.

Reply:
I think we should have a real default name. If there is too much output in the multi-model case, then the output in that case should be adjusted. No need to penalize the single-model case.

Reply:
That's a fair argument. Forcing model names to be unique identifiers will be helpful in the future for supporting interactions between models, but it will also make our output even more verbose.